tests/testing_support/validators/validate_log_events.py

```diff
@@ -79,6 +79,11 @@ def _check_log_attributes(expected, forgone_attrs, captured, mismatches):
                     mismatches.append("key: %s, value:<%s><%s>" % (key, value, captured_value))
                     return False
 
+        for key in required_attrs:
+            if not hasattr(captured, key) and key not in captured.attributes:
```

Comment:
This looks like it's checking both the attributes and directly on captured for the key. Why not do the same with the forgone_attrs check below for consistency?

Reply:
That's what the elif block did, but I moved some things around to make it maybe more readable.